lib/course.js

```diff
@@ -1,80 +1,113 @@
-let ContentWriter = require('./contentWriter');
-let yaml = require('js-yaml');
-let path = require('path');
+const yaml = require('js-yaml');
+const pathModule = require('path');
+const Content = require('./content');
+const { removeFalsy } = require('./helpers');
 
 /**
- * Given a course path, creates a nested Course object
+ * Given a course README string and a path, creates a nested Course object
+ * @param {string} content
  * @param {string} filepath
  * @class
- * @extends ContentWriter
+ * @extends Content
  */
-module.exports = class Course extends ContentWriter {
-  constructor (path) {
-    super(path)
-
+module.exports = class Course extends Content {
+  constructor({ body, path }) {
+    super({ body, path });
     this.topic = null;
     this.workouts = [];
     this.standards = [];
     this.sections = {};
     try {
       this.parse(this.rawText);
-    } catch(err) {
-      console.error(`Problem with ${this.contentPath}`)
+    } catch (err) {
+      console.error(`Problem with ${this.contentPath}`);
     }
   }
 
-/**
- * Recursively parses all child subtopics, workouts, and insights
- * @param {string} Course readme text
- */
+  /**
+   * Recursively parses all child subtopics, workouts, and insights
+   * @param {string} Course readme text
+   */
   parse(text) {
-    if (text.length == 0) throw new Error("Passed in empty readme text");
-    let parsed = yaml.safeLoad(text);
-    if (parsed.sections) {
-      parsed.sections.forEach((s) => {this.sections[s] = []});
-      if (parsed.sections.workouts) parsed.sections.forEach((s) => {s.workouts.forEach((w) => {this.sections[s].push(w)})})
+    if (text.length === 0) {
+      throw new Error('Passed empty Topic README');
+    }
+    yaml.safeLoadAll(text.split('---')[0], doc => {
+      for (const prop in doc) {
+        this[prop] = doc[prop];
+      }
+    });
+
+    if (this.next) {
+      this.next = this.next.map(entry => {
+        const [ topicSlug, subtopicSlug ] = entry.split(':');
+
+        return {
+          topic: topicSlug,
+          subtopic: subtopicSlug
```

Comment:
does the db schema use "subtopic" still?

Reply:
yeah our db models still use `subtopic`